zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/util/AdvancedSearchFilterUtilTest.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.util;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.camunda.search.filter.Operation;
+import io.camunda.search.filter.Operator;
+import io.camunda.zeebe.gateway.protocol.rest.AdvancedDateTimeFilter;
+import io.camunda.zeebe.gateway.protocol.rest.AdvancedIntegerFilter;
+import io.camunda.zeebe.gateway.protocol.rest.AdvancedLongFilter;
+import io.camunda.zeebe.gateway.protocol.rest.AdvancedStringFilter;
+import io.camunda.zeebe.gateway.protocol.rest.BasicLongFilter;
+import io.camunda.zeebe.gateway.rest.RestControllerTest;
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class AdvancedSearchFilterUtilTest {
+
+  private <F, T> F constructFilter(
+      final Class<F> fClass, final Class<T> pClass, final List<Operation<T>> operations)
+      throws Exception {
+    final var filter = fClass.getDeclaredConstructor().newInstance();
+    for (final Operation<T> op : operations) {
+      final var operator = op.operator();
+      final var methodName = "set$%s".formatted(StringUtils.capitalize(operator.getValue()));
+      final var existsName = "set$%s".formatted(StringUtils.capitalize(Operator.EXISTS.getValue()));
+      final var method =
+          switch (operator) {
+            case IN -> filter.getClass().getMethod(methodName, List.class);
+            case EXISTS, NOT_EXISTS -> filter.getClass().getMethod(existsName, Boolean.class);
+            default -> filter.getClass().getMethod(methodName, pClass);
+          };
+      method.setAccessible(true);
+      switch (operator) {
+        case IN ->
+            method.invoke(
+                filter,
+                op.values().stream()
+                    .map(v -> AdvancedSearchFilterUtil.convertValue(pClass, v))
+                    .toList());
+        case EXISTS, NOT_EXISTS -> method.invoke(filter, operator.equals(Operator.EXISTS));
+        default -> method.invoke(filter, AdvancedSearchFilterUtil.convertValue(pClass, op.value()));
+      }
+    }
+    return filter;
+  }
+
+  private static Stream<Arguments> provideAdvancedLongFilterParameters() {
```

Comment:
🔧 Since there are other types in here as well, should we just name this `provideAdvancedFilterParameters`